packages/video_player/video_player/test/video_player_test.dart

```diff
@@ -956,6 +956,41 @@ void main() {
       });
     });
   });
+  test('updates position', () async {
+    final VideoPlayerController controller = VideoPlayerController.networkUrl(
+      _localhostUri,
+      videoPlayerOptions: VideoPlayerOptions(),
+    );
+
+    await controller.initialize();
+
+    // Set a custom interval
+    const Duration customInterval = Duration(milliseconds: 100);
+
+    final List<Duration> positions = <Duration>[];
+    final Completer<void> intervalUpdateCompleter = Completer<void>();
+
+    // Listen for position updates
+    controller.addListener(() {
+      positions.add(controller.value.position);
+      if (positions.length >= 3 && !intervalUpdateCompleter.isCompleted) {
+        intervalUpdateCompleter.complete();
+      }
+    });
+    await controller.play();
+    for (int i = 0; i < 3; i++) {
+      await Future<void>.delayed(customInterval);
+      fakeVideoPlayerPlatform._positions[controller.textureId] =
+          Duration(milliseconds: i * customInterval.inMilliseconds);
+    }
+
+    // Wait for at least 3 position updates
+    await intervalUpdateCompleter.future.timeout(const Duration(seconds: 5));
+
+    // Verify that the intervals between updates are approximately correct
+    expect(positions[1] - positions[0], greaterThanOrEqualTo(customInterval));
```

Comment:
Wouldn't this test as written pass without the rest of the PR?

Reply:
just tested using 500 ms for the updates and it failed, but i will add another test related to the captions since its the more related to this pr

Reply:
I added the other test and I hope its shows the problem more, let me know if there anything else I can fix.

Thank you.